src/js/components/input/AutomagicFormiflyField.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import {useFormiflyContext} from '../meta/FormiflyContext';
+import FormiflyCheckField from './FormiflyCheckField';
+import FormiflyField from './FormiflyField';
+import FormiflyMultiSelectField from './FormiflyMultiSelectField';
+import FormiflyRadioGroup from './FormiflyRadioGroup';
+import FormiflySelectField from './FormiflySelectField';
+import {StyledComponent} from 'styled-components';
+
+const AutomagicFormiflyField = (props: AutomagicFormiflyFieldProps) => {
+    const {getFieldProps} = useFormiflyContext();
+    const fieldProps = getFieldProps(props.name, props.help, props.type, props.value, props.id, props.additionalDescribedBy);
+    const type = props.type ?? fieldProps.type;
+
+    if (!!props.options) {
```

Comment:
Could using `if (Boolean(props.options))` instead of double negation make TypeScript happier?

Reply:
unfortunately no. The best I could do is
```
const options = props.options;
    if (options !== undefined) {
        if (props.type === 'radio-group') {
            return <FormiflyRadioGroup {...fieldProps} {...props} options={options}/>;
```

not sure if this is much better. I'm not entirely sure why TS is complaining in the first place...

Reply:
We could change it to get all the props via object destructuring. That's how we write most of our components these days and IMO it is a bit prettier.

Not a necessary change but nice to have